dl/src/main/scala/com/intel/analytics/bigdl/dataset/Types.scala

```diff
@@ -89,6 +90,7 @@ class Sample[T: ClassTag] (
 
   def copyToFeature(storage: Array[T], offset: Int, length: Int): Unit = {
     require(offset + length <= storage.length, "index out of boundary")
+    require(length <= featureTensor.storage().array().length, "length too long for feature")
```

Comment:
copyToFeature will copy the class private variable feature out. It assumes the offset of featureTensor.storage always starts from 1, which is initialized by the copy() method.